javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/Utils.java

```diff
@@ -39,29 +42,31 @@ public static ClassLoader getAgentClassLoader() {
   public static BootstrapClassLoaderProxy getBootstrapProxy() {
     if (getAgentClassLoader() instanceof AgentClassLoader) {
       return ((AgentClassLoader) getAgentClassLoader()).getBootstrapProxy();
-    } else {
-      // in a unit test
-      return unitTestBootstrapProxy;
     }
+    // in a unit test
+    return unitTestBootstrapProxy;
   }
 
   /** com.foo.Bar to com/foo/Bar.class */
   public static String getResourceName(String className) {
-    if (!className.endsWith(".class")) {
-      return className.replace('.', '/') + ".class";
-    } else {
+    if (className.endsWith(".class")) {
       return className;
     }
+    return className.replace('.', '/') + ".class";
   }
 
   /** com/foo/Bar.class to com.foo.Bar */
   public static String getClassName(String resourceName) {
-    return resourceName.replaceAll("\\.class\\$", "").replace('/', '.');
+    return stripDotClassSuffix(resourceName).replace('/', '.');
   }
 
   /** com.foo.Bar to com/foo/Bar */
   public static String getInternalName(String resourceName) {
-    return resourceName.replaceAll("\\.class\\$", "").replace('.', '/');
+    return stripDotClassSuffix(resourceName).replace('.', '/');
+  }
+
+  private static String stripDotClassSuffix(String resourceName) {
+    return CLASS_SUFFIX_PATTERN.matcher(resourceName).replaceAll("");
```

Comment:
Maybe we can just do `endsWith` and `substring`?

Reply:
You mean `indexOf`+`substring`?

Reply:
Thought endsWith is ok since it's a fixed length so just subtract for the index. indexOf would end up searching the entire string.

Reply:
Yup, I like it!